src/AspNet.Security.OAuth.DingTalk/DingTalkAuthenticationHandler.cs

```diff
@@ -0,0 +1,227 @@
+            {
+                ["appid"] = Options.ClientId,
+                ["response_type"] = "code",
+                ["scope"] = FormatScope(),
+                ["redirect_uri"] = redirectUri,
+                ["state"] = stateValue
+            });
+        }
+
+        // See https://developers.dingtalk.com/document/app/obtain-orgapp-token for details.
+        protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OAuthCodeExchangeContext context)
+        {
+            var address = QueryHelpers.AddQueryString(Options.TokenEndpoint, new Dictionary<string, string?>
+            {
+                ["appkey"] = Options.ClientId,
+                ["appsecret"] = Options.ClientSecret
+            });
+            using var response = await Backchannel.GetAsync(address);
+            using var payload = await GetPayloadAsync(response);
+
+            return OAuthTokenResponse.Success(payload);
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            var code = Request.Query["code"];
+            var userInfoTmp = await GetUserInfoTemporaryAsync(code);
+            var userIdentifier = await GetUserIdentifierAsync(userInfoTmp.unionid!, tokens.AccessToken);
+
+            // See https://developers.dingtalk.com/document/app/query-user-details for details.
+            var address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, "access_token", tokens.AccessToken);
+            using var requestContent = new ReadOnlyMemoryContent(await CopyPayloadAsync(new Dictionary<string, Microsoft.Extensions.Primitives.StringValues>
+            {
+                ["userid"] = userIdentifier.userid,
+                ["language"] = Options.Language
+            }));
+            requestContent.Headers.ContentType = new MediaTypeHeaderValue(MediaTypeNames.Application.Json);
+            using var response = await Backchannel.PostAsync(address, requestContent, Context.RequestAborted);
+            using var payload = await GetPayloadAsync(response);
+
+            var result = payload.RootElement.GetProperty("result");
+
+            using var userInfo = JsonDocument.Parse(await MergePayloadAsync(result, userInfoTmp));
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, userInfo.RootElement);
+            context.RunClaimActions();
+
+            await Options.Events.CreatingTicket(context);
+            return new AuthenticationTicket(context.Principal!, context.Properties, Scheme.Name);
+        }
+
+        protected override string FormatScope() => string.Join(",", Options.Scope);
```

Comment:
Prefer using the char ',' instead of string ",".

Reply:
Done.